tests/unit/plugins/modules/test_jenkins_node.py

```diff
@@ -207,6 +207,45 @@ def test_state_present_when_absent_check_mode(get_instance, instance, state):
     assert result.value["changed"] is True
 
 
+@mark.parametrize(["state"], [param(state) for state in PRESENT_STATES])
+def test_state_present_when_absent_redirect_auth_error_handled(
+    get_instance, instance, state
+):
+    instance.node_exists.side_effect = [False, True]
+    instance.get_node_config.return_value = "<slave />"
+    instance.create_node.side_effect = jenkins.JenkinsException
+
+    set_module_args({
+        "name": "my-node",
+        "state": state,
+    })
+
+    with raises(AnsibleExitJson) as result:
+        jenkins_node.main()
+
+    assert instance.create_node.call_args == call("my-node", launcher=jenkins.LAUNCHER_SSH)
+
+    assert result.value["created"] is True
+    assert result.value["changed"] is True
+
+
+@mark.parametrize(["state"], [param(state) for state in PRESENT_STATES])
+def test_state_present_when_absent_other_error_raised(get_instance, instance, state):
+    instance.node_exists.side_effect = [False, False]
+    instance.get_node_config.return_value = "<slave />"
+    instance.create_node.side_effect = jenkins.JenkinsException
+
+    set_module_args({
+        "name": "my-node",
+        "state": state,
+    })
+
+    with raises(jenkins.JenkinsException):
```

Comment:
I would recommend catching that exception in the module code and "translating" it into `self.module.fail_json()` for consistency. Ansible will do something like that, but it is probably better to be explicit with this kind of thing.